queue_job/job.py

```diff
@@ -516,6 +532,9 @@ def store(self):
             db_record.write(vals)
         else:
             date_created = dt_to_string(self.date_created)
+            # We store the original context used at import on create
+            ctx = self.env.context.copy() or '{}'
+            vals.update({'original_context': json.dumps(ctx) or ''})
```

Comment:
@KKamaa I don't understand in which cases this would be different from `context`.

To me the code looks like this:

```
vals.update({"context": json.dumps(self.env.context.copy())}  # line 518
... some code that does not seem to modify `vals['context'] in any way
vals.update({"original_context": json.dumps(self.env.context.copy())}  # line 537
```

What am I missing? Can you give an example of a situation where `context` would end up different from `original_context` ?

Reply:
@thomaspaulb about the field  **original_context** I added it to just show how the context we are using changes in execution of the queue job. When we create the queue job we have the entire import context, however when we start executing it the context is lost/modified. So I save the execution from the beginning in **original_context** field, then from there use what was saved and current context execution the update to use when executing the queue.job so:
`{'params': {action: 23}} + {'lang': Fr_fr'} = {"params": {action: 23}, "lang': "Fr_fr"} for example`

about the _astor lib_ wasn't that successful with it and changed the part which was having issues. So if context contained a record say `job.batch(1,)` it was hard in conversion. So I changed where job_batch was being assigned the record class instead of the id. I had added samples on the issue.

So maybe we can just use context alone buy converting it the update it with current context then save again until the queue job is done. Just did a separate to have a proper sample test of the import working.